### PR TITLE
Add Keltner Channel Indicator

### PR DESCRIPTION
**Pull request by rideron89**
This pull request adds the Keltner Channel (KC) indicator. This indicator is very similar Bollinger Bands, with the exception that it uses Average True Range (ATR) instead of Standard Deviation (SD).

Some definitions of KC call for using an EMA, while others call for SMA. I went with EMA in this implementation because your ATR is calculated using EMA.

I'm rather new to Open Source and Math/Statistics are not my strong suits, so please feel free to be as critical of this request as necessary.

Thanks!!

---

- [https://en.wikipedia.org/wiki/Keltner_channel](https://en.wikipedia.org/wiki/Keltner_channel)
- [https://www.investopedia.com/terms/k/keltnerchannel.asp](https://www.investopedia.com/terms/k/keltnerchannel.asp)

### CODE REVIEW

**Comment by greyblake** [comment]
@rideron89 thanks for the PR.
Right now I am overloaded with work and personal stuff, but I'll try to review it this week.

File: src/indicators/bollinger_bands.rs [hunk 1]
```diff
@@ -42,7 +42,7 @@ use crate::{Close, Next, Reset};
 ///
 /// # Links
 ///
-/// ![Bollinger Bands, Wikipedia](https://en.wikipedia.org/wiki/Bollinger_Bands)
+/// * [Bollinger Bands, Wikipedia](https://en.wikipedia.org/wiki/Bollinger_Bands)
```
Comment (greyblake) [inline src/indicators/bollinger_bands.rs:45]: thanks!

File: src/indicators/keltner_channel.rs [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+
+        Self::Output {
+            average,
+            upper: average + atr * self.multiplier,
+            lower: average - atr * self.multiplier,
+        }
+    }
+}
+
+impl<'a, T: Close> Next<&'a T> for KeltnerChannel {
```
Comment (greyblake) [inline src/indicators/keltner_channel.rs:99]: I think here, according to the formula in wikipedia, we should use `Close`, `High`, `Low`.

It's a rare indicator, where implementation of `next<f64>` and `next<T>` are different.
For `next<T>` we'd like:
* `ema.next((input.close + input.high + input.low) / 3.0)`
* `atr.next(input)`

Note: ATR has also 2 alternative implementations for `next<f64>`  and `next<T>`

**Comment by greyblake** [comment]
@rideron89 Hi, sorry for my long inactivity.
Thank you a lot for the nicely prepared pull requested.
I've got a little question regarding calculating average and ATR when input contains close,high,low.
Apart from that the PR looks great and I'd like to to merge once we resolve that question.

**Comment by greyblake** [comment]
@rideron89 I'll merge the PR and will adjust the implementation slightly.
Thanks again!

**Comment by rideron89** [comment]
That works!

Yeah I think I agree with your comments, so I'll look forward your finished product.

Thanks!

**Comment by greyblake** [comment]
@rideron89 Keltner Channel is available in just released version 0.2.0.
Again sorry for the very long delay.

**Pull request merged**